tests/datatypes/test_tuple.py

```diff
@@ -76,6 +76,14 @@ def test_lt(self):
                     print(p1, "<", p2, ":", p1 < p2)
             """, run_in_function=False)
 
+    def test_immutable(self):
+        self.assertCodeExecution("""
+            x = (1, 2)
+            x[0] = 3
+            x[-1] = 3
+            x[2] = 3
+            """, run_in_function=False)
```

Comment:
Right, I was checking Python's behaviour in the interactive console, so it didn't register that execution would stop because an exception is raised.

I've just pushed a commit to add separate try-except blocks for each line, and print the error out to make sure the error text matches too. I'm not sure if there's a more elegant way to do this...